File: Backend/demo/src/main/java/com/example/Controllers/AddOnMasterController.java
package com.example.Controllers;

import java.util.List;
import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import com.example.Models.AddOnMaster;
import com.example.Services.AddOnMasterService;

@RestController
@RequestMapping("/api/addons")
@CrossOrigin(origins = "http://localhost:3000")
public class AddOnMasterController {

    @Autowired
    private AddOnMasterService addOnMasterService; 


    @GetMapping
    public ResponseEntity<List<AddOnMaster>> getAllAddOns() {
        List<AddOnMaster> addOns = addOnMasterService.getAllAddOns();
        if (addOns.isEmpty()) {
            return new ResponseEntity<>(HttpStatus.NO_CONTENT); 
        }
        return new ResponseEntity<>(addOns, HttpStatus.OK); 
    }

    
    @GetMapping("/{addonId}")
    public ResponseEntity<AddOnMaster> getAddOnById(@PathVariable Long addonId) {
        AddOnMaster addOnMaster = addOnMasterService.getAddOnById(addonId);
        if (addOnMaster == null) {
            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
        }
        return new ResponseEntity<>(addOnMaster, HttpStatus.OK); 
    }

    
    @DeleteMapping("/{addonId}")
    public ResponseEntity<String> deleteAddOnById(@PathVariable Long addonId) {
        AddOnMaster addOnMaster = addOnMasterService.deleteAddOnById(addonId);
        if (addOnMaster == null) {
            return new ResponseEntity<>("AddOn not found", HttpStatus.NOT_FOUND); 
        }
        return new ResponseEntity<>("AddOn deleted successfully", HttpStatus.OK); 
    }

    @PutMapping("/{addonId}")
    public ResponseEntity<AddOnMaster> updateAddOn(
            @PathVariable Long addonId,
            @RequestBody AddOnMaster updatedAddOn) {
        AddOnMaster updated = addOnMasterService.updateAddOn(addonId, updatedAddOn);
        if (updated == null) {
            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
        }
        return new ResponseEntity<>(updated, HttpStatus.OK); 
    }

    @PostMapping
    public ResponseEntity<AddOnMaster> createAddOn(@RequestBody AddOnMaster addOnMaster) {
        AddOnMaster createdAddOn = addOnMasterService.createAddOn(addOnMaster); 
        return new ResponseEntity<>(createdAddOn, HttpStatus.CREATED);     }
}

File: Dot net Backend/latest/Fleet_Management/Controllers/AddOnMasterController.cs
using Fleet_Management.Exceptions;
using Fleet_Management.Models;
using Fleet_Management.Service;
using Microsoft.AspNetCore.Mvc;

namespace Fleet_Management.Controllers
{
    [Route("api/addons")]
    [ApiController]
    public class AddOnMasterController : ControllerBase
    {
        private readonly IAddOnMasterService _addOnMasterService;

        public AddOnMasterController(IAddOnMasterService addOnMasterService)
        {
            _addOnMasterService = addOnMasterService;
        }


        [HttpGet]
        public async Task<ActionResult<IEnumerable<AddOnMaster>>> GetAllAddOns()
        {
            var addOns = await _addOnMasterService.GetAllAddOnsAsync();
            if (addOns.Count == 0)
            {
                return NoContent();
            }
            return Ok(addOns);
        }


        [HttpGet("{addonId}")]
        public async Task<ActionResult<AddOnMaster>> GetAddOnById(long addonId)
        {
            var addOnMaster = await _addOnMasterService.GetAddOnByIdAsync(addonId);
            if (addOnMaster == null)
            {
                return NotFound();
            }
            return Ok(addOnMaster);
        }


        [HttpDelete("{addonId}")]
        public async Task<ActionResult> DeleteAddOnById(long addonId)
        {
            var addOnMaster = await _addOnMasterService.DeleteAddOnByIdAsync(addonId);
            if (addOnMaster == null)
            {
                return NotFound("AddOn not found");
            }
            return Ok("AddOn deleted successfully");
        }


        [HttpPut("{addonId}")]
        public async Task<ActionResult<AddOnMaster>> UpdateAddOn(long addonId, [FromBody] AddOnMaster updatedAddOn)
        {
            if (addonId != updatedAddOn.AddonId)
            {
                return BadRequest("AddonId in URL does not match AddonId in the body");
            }

            var updated = await _addOnMasterService.UpdateAddOnByIdAsync(updatedAddOn);
            if (updated == null)
            {
                return NotFound();
            }
            return Ok(updated);
        }


        [HttpPost]
        public async Task<ActionResult<AddOnMaster>> CreateAddOn([FromBody] AddOnMaster addOnMaster)
        {
            var createdAddOn = await _addOnMasterService.CreateAddOnAsync(addOnMaster);
            return CreatedAtAction(nameof(GetAddOnById), new { addonId = createdAddOn?.AddonId }, createdAddOn);
        }

        // [HttpGet("throw")]
        // public IActionResult ThrowException()
        // {
        //     throw new ApiException("Test Exception for Global Middleware");
        // }
    }
}
